docs/usecases/testing.md

```diff
@@ -72,3 +72,202 @@ Test aspects are powerful tools for modifying behavior of individual tests or ev
 **Zero Dependencies**
 
 As a library with zero third party dependencies, this project is available on the JVM, ScalaJS, and will be available on Dotty and Scala Native in the near future. So you can write your tests once and make sure that your code works correctly across all platforms that you support.
+
+
+_Using Mocks in tests_
+
+What we expect from tests (at least those that we consider unit tests) is to be stable i.e. consecutive runs should yield the same results and take
+more or less the same amount of time. Biggest source of complexity during testing comes from external services which we cannot control like external
+payment APIs, object storages, http APIs ect. It is normal to hide these kind of services behind an interface and provide mock implementation to regain
+control and determinism. However there is another source of complexity that comes from the local infrastructure that is also hard to control without building prior abstractions. Things like stdin/stdout, clocks, random generators, schedulers can make writing tests hard or even impossible. Fortunately ZIO abstracted most of it in its runtime under `Environment` type. Thanks to this design `zio-test` could easily provide its own implementation named `MockEnvironment` which gives you mocks of mentioned infrastructure. In most of the cases when you'll be using `ZIO`s `testM` mocks are already created and should be controlled by exposed functions on companion object. If for some reason you would like to provide custom environment or are using other testing framework but still want to use mocks there are `make` functions on companion objects of mocks where you can construct your own. 
+
+A mistake that is easy to do is to create new instance of mock but use the one provided by the `MockEnvironment`. 
```

Comment:
Maybe say accidentally use different mock instances at the same time?

Reply:
🆗